### PR TITLE
docs: Add new tutorials 4a, 4b, 4c and 5

### PR DESCRIPTION
**Pull request by lzpap**
## Description
- Add `4.a Generate Address` tutorial to generate a random seed and an address from it, furthermore to load up the owned address with devnet tokens.
- Add `4.b Check Balance` tutorial to check the balance of the address from the previous tutorial.
- Add `4.c Get Account Data` tutorial to alternatively check the balance from 4.a.
- Add `5. Send Tokens` tutorial to send a value transfer to an arbitrary address.

### CODE REVIEW

File: docs/tutorials.rst [hunk 1]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
 the attribute contains a signature rather than a message.
 
+4.a Generate Address
+--------------------
+
+In this example, you will learn how to:
+
+- **Generate a random seed.**
+- **Generate an IOTA address that belongs to your seed.**
+- **Acquire free devnet IOTA tokens that you can use to play around with.**
```
Comment (todofixthis) [inline docs/tutorials.rst]: ❔ devnet or testnet?  I think in earlier tutorials we refer to it as testnet.

File: docs/tutorials.rst [hunk 2]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+Since we don't need to generate an address, there is no need for a seed to be
+employed in the API object. Note the ``time`` import, we need it for later.
+
+.. literalinclude:: ../examples/tutorials/04b_check_balance.py
+   :lines: 10-25
+   :lineno-start: 10
+
+Our script will poll the network for the address balance as long as the returned
+balance is zero. Therefore, the address you declared as ``my_address`` should
+have some balance. If you see the ``Zero balance found,..`` message couple times,
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
have some balance. If you see the ``Zero balance found...`` message a couple of times,
```

File: docs/tutorials.rst [hunk 3]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+
+We will need ``pprint`` for a prettified output of the response ``dict`` and
+``time`` for polling until we find non-zero balance.
+
+.. literalinclude:: ../examples/tutorials/04c_get_acc_data.py
+   :lines: 5-13
+   :lineno-start: 5
+
+Copy your seed from `4.a Generate Address`_ onto line 6. The API will use your
+seed to generate addresses an look for corresponding transactions on the Tangle.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
seed to generate addresses and look for corresponding transactions on the Tangle.
```

File: docs/tutorials.rst [hunk 4]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+.. note::
+
+    If you read :py:meth:`~Iota.get_account_data` documentation carefully, you
+    notice that you can gain control over which addresses are checked during
+    the call by specifying the ``start`` and ``stop`` index parameters.
+
+    This can be useful when your addresses with funds do not follow each other
+    in the address namespace, or a snapshot removed transactions from the
+    Tangle. It is recommended that you keep a local database of your already
+    used address indices.
```
Comment (todofixthis) [inline docs/tutorials.rst]: ❔ Include a link to account module here?

File: docs/tutorials.rst [hunk 5]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+Put your seed from `4.a Generate Address`_ onto line 4. We pass this seed to
+the API object, that will utilize it for signing the transfer.
+
+.. literalinclude:: ../examples/tutorials/05_send_tokens.py
+   :lines: 13-16
+   :lineno-start: 13
+
+In IOTA, funds move accross addresses, therefore we need to define a **receiver
+address**. For value transfers, you might only use addresses that are generated
+from seeds, so a randomly generated address doesn't qualify here. Re-run
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```rst
For testing value transfers, you should send the funds only to addresses that you control; if you use a randomly-generated receiver address, you won't be able to recover the funds afterward!
```

File: docs/tutorials.rst [hunk 6]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+`4.a Generate Address`_ for a new seed and a new address, or just paste a
+valid IOTA address onto line 16.
+
+.. literalinclude:: ../examples/tutorials/05_send_tokens.py
+   :lines: 18-25
+   :lineno-start: 18
+
+We declare a :py:class:`ProposedTransaction` object like we did before, but
+this time, with ``value=1`` parameter. The smallest value you can send is 1
+iota, there is no way to break it into smaller chunks. It is a really small
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
iota ("1i"); there is no way to break it into smaller chunks. It is a really small
```

File: docs/tutorials.rst [hunk 7]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+- Gathering ``inputs`` (addresses you own and have funds) to fund the 1i transfer.
+- Generating a new ``change_address``, and automatically sending the remaining
+  funds (``balance of chosen inputs`` - 1i) from ``inputs`` to ``change_address``.
+
+    .. warning::
+
+        This step is extremely important, as it prevents you from `spending twice
+        from the same address`_.
+
+        When an address is used an input, all tokens will be withdrawn. Part of
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
        When an address is used as an input, all tokens will be withdrawn. Part of
```

File: docs/tutorials.rst [hunk 8]
```diff
@@ -228,7 +232,278 @@ method to drop values we can't decode using ``utf-8``, or if the raw trytes
...
+- Doing proof-of-work for each transaction in the bundle and sending it to the
+  network.
+
+
+.. literalinclude:: ../examples/tutorials/05_send_tokens.py
+   :lines: 31-32
+   :lineno-start: 31
+
+Open the link and observe the bundle you have just sent to the Tangle. Probably
+it will take couple of seconds for the network to confirm it.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
it will take a couple of seconds for the network to confirm it.
```

File: docs/tutorials.rst (hunk 1 above)
Comment (lzpap) [reply docs/tutorials.rst]: The official docs call it [devnet](https://docs.iota.org/docs/getting-started/0.1/network/iota-networks), so I created this ticket #263. Planned to be in the next release.

**Pull request merged**